### ISSUE TITLE
tW Tool Rewrite with new architecture

### ISSUE THREAD

**Issue by benjore** [unfoldingWord/translationCore#4318]

**Comment by RoyalSix** [comment]
- [x] First merging Joels PR (After reviewing)
https://github.com/unfoldingWord-dev/translationCore/pull/4489
After this wordAlignment will be working and the new `tc-tool` will be in the core.

- [x] Then we need to review and merge all the PR's for the `tc-ui-toolkit` and change the package.json in tW to actually have the module installed https://github.com/translationCoreApps/translationWords/blob/658ea4eec6bd92d61d3b74cbc39cbe6c75552be9/package.json#L67

- [x] Then we will need to remove all the old code from tW that is n already coming from the ui-toolkit. 

- [x] Then we will need to incorporate the changes Joel made in order to receive the correct data for tW to work again (i.e. wordAlignmentReducer from the API of `tc-tool`) 

- [x] Then we will need to either A. Have the strings in the ui-toolkit (And do it the right way later), or B. Expose the strings from the tool to the toolkit in each component.

- [x] Then we can make a final PR to `translationCore` (Some things from the original PR in 4492 may not be needed after incorporating Joels changes, forsure somethings will though such as removing the GroupMenu ) and `translationWords` repos and everything should be working.

**Comment by cckozie** [comment]
0.10.0 (45ccd50)
There are a number of problems with the new tW implementation:
1. <s>The Menu/Filter bar at the top of the group menu now scrolls off the screen when the group menu is scrolled down.</s>
![image](https://user-images.githubusercontent.com/8171759/41128432-c4ed5e68-6a73-11e8-8328-bfb89423d2c2.png)
2. <s>The blue used to signify percentage complete is now much darker making it hard to distinguish from the black background.</s>
new:
![image](https://user-images.githubusercontent.com/8171759/41128501-10101aa2-6a74-11e8-8d8d-2f029a2e3b50.png)
old:
![image](https://user-images.githubusercontent.com/8171759/41128473-e7b8fee8-6a73-11e8-9eeb-d56bb1b42d39.png)
3. <s>The thelps panel no longer is automatically displayed when the tool is opened.</s>
4. **[Moved to #4590]** ~~The 'See more' link is now visible in black text when the thelps panel is open~~
![image](https://user-images.githubusercontent.com/8171759/41128625-749dd2ca-6a74-11e8-8d6a-23c16def2af5.png)
5. <s>Nothing appears to be localized except the words 'Home' and 'Actions' on the menu bar</s>
6. <s>The Bookmark switch is no longer left aligned on the form</s>
7. <s>When the text in the check pane is expanded, the book name appears above the close 'X' rather than in the center of the title bar.</s>
![image](https://user-images.githubusercontent.com/8171759/41129102-375c7392-6a76-11e8-82a6-c6c4b3c5b119.png)
8. <s>When the tHelps pane is expanded, the 'C' on the close button is not capitalized</s>
9. <s>When the tHelps pane is expanded, the wrong content is sometimes displayed. To replicate:</s>
a. Open a Titus project in tW
b. Open the tHelps pane
c. Click on 'Christ, Messiah' in the group menu
d. Expand the tHelps pane
e. Click on the '(See also: How to Translate Names)' link
f. Click the Close button
g. Click on 'faith' in the group menu
h. Expand the tHelps pane
Notice that the article displayed is still How to Translate Names
10. There is no header on the Target Lane card in the expanded scripture pane (image below)
11. **[Moved to #4593]** ~~In the expanded scripture pane, the background color of spaces in even numbered verses is white on all translations except the target language (image below)~~
12. **[Moved to #4592]** ~~In the expanded scripture pane, the height of the header row is much greater than it needs to be~~
![image](https://user-images.githubusercontent.com/8171759/41129696-dfd3558e-6a78-11e8-9bd2-dbe800aabad3.png)


I have not finished testing the refactored tW tool, but am stopping here until Monday.

**Comment by RoyalSix** [comment]
## New Tasks
- [x] The Menu/Filter bar at the top of the group menu now scrolls off the screen when the group menu is scrolled down.
- [x] The blue used to signify percentage complete is now much darker making it hard to distinguish from the black background.
new
- [x] The thelps panel no longer is automatically displayed when the tool is opened.
- [x] Fix tW Localization
- [x] The Bookmark switch is no longer left aligned on the form
- [x] When the text in the check pane is expanded, the book name appears above the close 'X' rather than in the center of the title bar.
- [x] When the tHelps pane is expanded, the 'C' on the close button is not capitalized
- [x] When the tHelps pane is expanded, the wrong content is sometimes displayed.
- [x] There is no header on the Target Lane card in the expanded scripture pane
- [ ] Scripture pane verse editor not being reset for new check

**Comment by RoyalSix** [comment]
For #10 I am not able to replicate, can you share the project you used? @cckozie 
@RoyalSix See [en_ult_tit_text.zip](https://github.com/unfoldingWord-dev/translationCore/files/2094439/en_ult_tit_text.zip) It was imported from USFM.

**Comment by RoyalSix** [comment]
Ok @cckozie I investigated that issue and it is unrelated to the changes made here. That is project problem. 
That same thing happens if you use that project in 0.10.0 (before the refactor) 
@RoyalSix Unfortunately it's not a problem with the project. Any project imported from USFM has the same issue. I'll create a new issue for it.

**Comment by richmahn** [comment]
@RoyalSix Getting this big white 0 to the right of the funnel in the Menu header:

![2018-06-14](https://user-images.githubusercontent.com/2839925/41441979-5f58fda8-6ff2-11e8-9d56-922f38ff17c4.png)

**Comment by richmahn** [comment]
@RoyalSix No links work in the tHelps when it is just a sidebar, they do work when tHelps is expanded to a modal.

**Comment by cckozie** [comment]
0.10.0 (ed1e141)
1. <s>When a project is loaded in tW, it should open in 'Select' mode. The user should not have to click the Select button before making a selection.</s>
https://github.com/translationCoreApps/translationWords/pull/136
2. <s>When in Select mode, the Cancel, Clear, and Save buttons should be right aligned. The Save button should be located in the same place as the Save & Continue button so the user doesn't need to move the cursor between clicks of those buttons.</s>
https://github.com/translationCoreApps/tc-ui-toolkit/pull/42
3. <s>The Clear Selection button should be enabled as soon as any selection has been made rather than only after the selection has been Saved.</s>
https://github.com/translationCoreApps/tc-ui-toolkit/pull/43
4. <s>When in Edit or Comment mode, the Cancel and Save buttons should be right aligned.</s>
https://github.com/translationCoreApps/tc-ui-toolkit/pull/42
NOTE: I am still testing this issue, so there may be more items added here.

**Comment by cckozie** [comment]
All issues appear to have been fixed in 0.10.0 (2270c71), other than #10 noted above which has been moved to #4394.
